roofit/roofitcore/inc/RooFuncWrapper.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Project: RooFit
+ * Authors:
+ *   Garima Singh, CERN 2022
+ *
+ * Copyright (c) 2022, CERN
+ *
+ * Redistribution and use in source and binary forms,
+ * with or without modification, are permitted according to the terms
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)
+ */
+
+#ifndef RooFit_RooFuncWrapper_h
+#define RooFit_RooFuncWrapper_h
+
+#include "TROOT.h"
+#include "TSystem.h"
+#include "RooAbsReal.h"
+#include "RooGlobalFunc.h"
+#include "RooMsgService.h"
+#include "RooRealVar.h"
+#include "RooListProxy.h"
+
+#include <memory>
+#include <string>
+
+/// @brief  A wrapper class to store a C++ function of type 'double (*)(double* )'.
+/// @tparam Func Function pointer to the generated function.
+template <typename Func = double (*)(double *)>
+class RooFuncWrapper final : public RooAbsReal {
+public:
+   RooFuncWrapper(std::string funcName, std::string funcBody, RooArgSet const &paramSet)
+      : RooAbsReal{"RooFuncWrapper", "RooFuncWrapper"}, _paramProxies{"!params", "List of parameters", this},
+        _funcName(funcName)
+   {
+      bool comp = gInterpreter->Declare(funcBody.c_str());
+      if (!comp) {
+         std::stringstream errorMsg;
+         errorMsg << "Function " << funcName << " could not be compiled. See above for details.";
+         coutE(InputArguments) << errorMsg.str() << std::endl;
+         throw std::runtime_error(errorMsg.str().c_str());
+      }
+      _func = (Func)gInterpreter->ProcessLine((funcName + ";").c_str());
+      for (auto *param : paramSet) {
+         _paramProxies.add(*param);
+         if (auto realParam = dynamic_cast<RooRealVar *>(param))
+            _params.emplace_back(realParam);
+      }
```

Comment:
```suggestion
         if (!dynamic_cast<RooRealVar *>(param))
            // Print error message and throw exception like above.
      }
```
Instead of filling the redundant `_params` data member, you should better check that all the types are `RooAbsReal*`. It doesn't have to be RooRealVar, which is less general (a RooRealVar is a specialization of a RooAbsReal where you can set the value manually with setVal().